docs/DeveloperGuide.md

```diff
@@ -1,34 +1,140 @@
+The Module parsing logic component retrieves module information from NUSMODS API.
+
+**Structure of the module logic component:**
+<br>
+The module parsing logic of our application consists of 4 classes: ```TimetableParser ModuleApiParser ModuleHandler LessonsGenerator```.
+A detailed implementation would be explained in the implementation section.
+<br>
+ API: ```logic.modulelogic.LessonsGenerator```
+<br>
+1. ```LessonsGenerator``` uses the ```TimetableParser``` class to acquire the modules a user is taking, including the timeslots of those modules.
+2. ```LessonsGenerator``` also uses ```Modulehandler``` to retrieve a set of information related to a specific module.
+3. With both information, ```LessonsGenerator``` is able to dynamically generate the user's time-slots stored in ```ArrayList<String[]>``` via a series of Key-Value pair hashing.
+4. ```Arraylist<String[]> ``` contains the start/end time, days and weeks of all modules the user is taking.
+<br>
+
+### 2.4. Schedule component
+The Schedule component finds common free slots from team members’ schedules.
+
+**Structure of the Schedule component:**
+
+The Schedule component of our application consists of 3 classes ```TeamMember```, ```TeamMemberList``` and ```ScheduleHandler```. 
+
+1. ```TeamMember``` consists of information of a member's schedule.
+2. ```TeamMemberList``` is a ```Arraylist<TeamMember> ``` which new ```TeamMember``` can be added to.
+3. ```ScheduleHandler``` retrieves the schedule of selected ```TeamMember``` in the ```TeamMemberList```, to generate a combined schedule.
+### 2.5. Meeting component
+The Meeting component of our application consists of 2 classes: ```Meeting MeetingList```
+<br>
+
+### 2.6. Exception classes
+
+### 2.7. Common classes
+
+
+## 3. Implementation
+This section describes some noteworthy details of how our application works in the backend.
+### 3.1. Detailed implementation of logic.modulelogic component
+![logic.modulelogic Component](images/TimetableParser.png)<br>
+The above figure shows```TimetableParser```, a private class called exclusively by ```LessonsGenerator```. It makes use of regex to sift through timetable link provided by user in the form of ```String``` object and stores
+the semester and the user's module information according to the timetable link provided. It depends on the ```common.Messages``` class to provide the exception message when an incorrect link is being parsed.<br>
+![logic.modulelogic Component](images/ModuleHandler.jpg)<br>
+ From the figure above, ```ModuleApiParser``` instantiates a HTTP GET request object to fetch a Json object from the open-sourced NUSMOD API server, and is called by ```ModuleHandler``` every time a particular module information is requested.
+ <br>
+```ModuleHandler``` cleans the data provided by ```ModuleApiParser``` and stores an easy to use data structure to be used by ```LessonsGenerator```.
+![logic.modulelogic Component](images/LessonsGenerator.jpg)<br>
```

Comment:
![dg_review2](https://user-images.githubusercontent.com/38548968/77999984-1dada880-7366-11ea-9dc9-4525d424c37d.PNG)

Should the creation of a new LessonsGenerator point directly to the LessonsGenerator class since it is creating a new LessonsGenerator object?